### PR TITLE
Refactor pipelines to remove duplicate code

### PR DESCRIPTION
**Pull request by ocallesp**
We are only making changes to these two pipelines:
- https://github.com/dotnet/try/blob/main/azure-pipelines-CI.yml
- https://github.com/dotnet/try/blob/main/azure-pipelines.yml

This part of [Consolidate fixes in our yml pipeline](https://github.com/dotnet/try/issues/1150)

### CODE REVIEW

File: eng/templates/setup-template.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# setup-template.yml
```
Comment (AbhitejJohn) [inline eng/templates/setup-template.yml:1]: Could we add a comment to say that this affects both windows and linux builds here please? It should be apparent from code but wouldn't hurt to have multiple indicators.

**Comment by AbhitejJohn** [comment]
Was this primarily re-arranging existing tasks or did we have to add/remove tasks as well?

**Review by AbhitejJohn — APPROVED** [review]
LGTM, will let Dale/Jon who have been spending time here take a stab at a review as well though.

**Comment by dalehhirt** [comment]
Talked with Osvaldo, he is working on updating this

**Pull request merged**